gradle-baseline-java-config/resources/checkstyle/checkstyle.xml

```diff
@@ -269,6 +280,42 @@
             <property name="format" value="^import org\.apache\.commons\.math\."/>
             <property name="message" value="math is deprecated, use math3 instead"/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="\bPreconditions\.check.*\{\}"/>
+            <property name="message" value="Use printf-style formatting in Preconditions"/>
+            <property name="ignoreComments" value="true"/>
+        </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="\.(debug|info|warn|error)\(.*%[sd]"/>
```

Comment:
This is the one I feel strongest about - very common to see people get this wrong and nobody seems to test their error messages.  Happy to take advice on the regex...
The one we had on Phoenix was actually: `\.(debug|info|warn|error)\(&quot;[^&quot;%]*%[^&quot;%]*&quot;(,|\))`
An option that is more prone to false negatives but still likely to catch problems would be: `log(ger)?\.(debug|info|warn|error)\(.*%[sd]`